src/aks-preview/azext_aks_preview/custom.py

```diff
@@ -1012,7 +1014,13 @@ def aks_scale(cmd,  # pylint: disable=unused-argument
                 raise CLIError(
                     "Cannot scale cluster autoscaler enabled node pool.")
 
-            agent_profile.count = int(node_count)
+            if agent_profile.type == CONST_VIRTUAL_MACHINES:
+                if len(agent_profile.virtual_machines_profile.scale.manual) == 1:
+                    agent_profile.virtual_machines_profile.scale.manual[0].count = int(node_count)
+                else:
+                    raise CLIError("Cannot scale virtual machines node pool with more than one size.")
```

Comment:
Better use a more specific error type listed in `azure.cli.core.azclierror` ,e.g., `InvalidArgumentValueError`

Reply:
change to ClientRequestError now

Reply:
change to ClientRequestError now